src/HealthGPS.Console/model_parser.cpp

```diff
@@ -203,14 +208,17 @@ EnergyBalanceModelDefinition load_newebm_risk_model_definition(const host::poco:
         }
     }
 
-    return EnergyBalanceModelDefinition(std::move(nutrient_list), std::move(nutrient_equations));
+    return std::make_shared<EnergyBalanceModelDefinition>(std::move(nutrient_list),
+                                                          std::move(nutrient_equations));
 }
 
 void register_risk_factor_model_definitions(CachedRepository &repository, const ModellingInfo &info,
                                             const SettingsInfo &settings) {
     MEASURE_FUNCTION();
+    HierarchicalModelType model_type;
+    std::shared_ptr<RiskFactorModelDefinition> model_definition;
```

Comment:
In this case, they're eventually stashed in a vector of shared_ptr base models, to allow polymorphic model definition storage in `CachedRepository`. I can't use `unique_ptr` in a vector without having to `std::move` the definition out of it later, defeating the point of the cache. I mostly agree that the cache is probably unnecessary, and should be removed in future though. I suppose I could start as a `unique_ptr` and then `std::move` into the cache vector once it's done.

Reply:
Ah, I see. Well in that case I guess you will need to copy it around. Like I say, definitely not a biggie anyways.

Have you worked out what the cache is for btw? I can't really think what would need caching... Does the code reload the config file in various places or something?

Reply:
I think its original purpose was to load backend filesystem data, and cache it to avoid reloading, but it seems to have been hijacked to pass around model definitions everywhere and for the module factory (yet another reason to do away with this module factory business in HealthGPS.Console). Its use as a filesystem (or database?) cache is still valid though. Just needs a tidy-up.

Reply:
Ahh, I see. That makes sense.